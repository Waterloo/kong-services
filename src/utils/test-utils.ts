import { createRouter, createMemoryHistory } from 'vue-router'

/**
 * Creates a test router with the application's routes.
 * Useful for component tests that require routing functionality.
 */
export function createTestRouter() {
  return createRouter({
    history: createMemoryHistory(),
    routes: [
      {
        path: '/',
        name: 'home',
        component: { template: '<div>Home</div>' },
      },
      {
        path: '/:id',
        name: 'serviceDetails',
        component: { template: '<div>Details</div>' },
      },
    ],
  })
}
