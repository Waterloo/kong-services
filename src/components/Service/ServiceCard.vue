<template>
  <KCard
    :aria-label="`Open ${service.name}`"
    class="service-card"
    role="button"
    tabindex="0"
  >
    <template #title>
      <div class="card-header">
        <div class="status-badge">
          <component
            :is="statusConfig.icon"
            :color="statusConfig.color"
          />
          <span class="status-text">{{ statusConfig.text }}</span>
        </div>
        <KBadge appearance="info">
          {{ service.versions.length }} Versions
        </KBadge>
      </div>
    </template>

    <div class="card-content">
      <h1 class="service-title">
        {{ service.name }}
      </h1>
      <p class="service-description">
        {{ service.description }}
      </p>
    </div>

    <template #footer>
      <div
        v-if="service.configured"
        class="additional-info"
      >
        <div class="metrics">
          <div class="metric-row">
            <KMetric
              label="latency"
              :value="service.metrics?.latency ?? 0"
            />
          </div>
          <div class="metric-row">
            <KMetric
              label="uptime"
              :value="(service.metrics?.uptime.toFixed(2) ?? 0) + '%'"
            />
          </div>
          <div class="metric-row">
            <KMetric
              label="requests"
              :value="formatCompactNumber(service.metrics?.requests ?? 0)"
            />

            <KMetric
              label="errors"
              :value="(service.metrics?.errors?.toFixed(2) ?? 0) + '%'"
            />
          </div>
        </div>
        <div class="service-avatars">
          <KAvatar
            v-for="avatar in avatars"
            v-bind="avatar"
            :key="avatar.key"
          />
        </div>
      </div>

      <div
        v-else
        class="not-configured-message"
      >
        <span class="message-text"><span class="dot" /> Not configured with runtime yet</span>
      </div>
    </template>
  </KCard>
</template>

<script setup lang="ts">
import { KCard, KBadge } from '@kong/kongponents'
import KMetric from '../shared/KMetric.vue'
import type { Service } from '@/types/models/services'
import { computed } from 'vue'
import { getStatusConfig } from '@/utils/services'
import KAvatar from '../shared/KAvatar.vue'
import { formatCompactNumber } from '@/utils/number'

interface Props {
  service: Service
}

const props = defineProps<Props>()

const statusConfig = computed(() =>
  getStatusConfig(props.service.published, props.service.configured),
)

const avatars = computed(() => {
  const avatars = [...new Set(
    props.service.versions
      .map(v => v.developer?.avatar)
      .filter(Boolean),
  )]

  return avatars.length <= 2
    ? avatars.map(src => ({ src, key: src }))
    : [
      { label: `+${avatars.length - 2}`, key: 'more-avatars' },
      ...avatars.slice(0, 2).map(src => ({ src, key: src })),
    ]
})
</script>

<style lang="scss" scoped>


.service-card {
  border: none;
  border-radius: $kui-border-radius-10;
  gap: $kui-space-50;
  max-width: 880px;
  width: 100%;

  :deep(.k-card-header) {
    padding: $kui-space-80;
    padding-bottom: 0;
  }

  :deep(.k-card-body) {
    padding: $kui-space-80;
    padding-bottom: 0;
    padding-top: $kui-space-80;
  }

  :deep(.k-card-actions) {
    border-top: $kui-border-width-10 solid $kui-color-border;
    padding: $kui-space-80;
    padding-top: $kui-space-60;
  }
}

.card-header {
  align-items: center;
  display: flex;
  justify-content: space-between;
}

.status-badge {
  align-items: center;
  color: $kui-color-text-neutral-strong;
  display: flex;
  font-size: $kui-font-size-30;
  font-weight: $kui-font-weight-regular;
  gap: $kui-space-40;
  line-height: $kui-line-height-30;
}

.status-text {
  color: $kui-color-text-neutral-strong;
}

.card-content {
  display: flex;
  flex-direction: column;
}

.service-title {
  color: $kui-color-text-neutral-stronger;
  font-family: $kui-font-family-heading;
  font-size: $kui-font-size-70;
  font-weight: $kui-font-weight-semibold;
  letter-spacing: $kui-letter-spacing-minus-40;
  line-height: $kui-line-height-70;
  margin-bottom: $kui-space-50;
  margin-top: $kui-space-0;
}

.service-description {
  color: $kui-color-text-neutral;
  font-size: $kui-font-size-40;
  font-weight: $kui-font-weight-regular;
  line-height: $kui-line-height-40;
  margin-top: $kui-space-0;
}

.additional-info {
  align-items: center;
  display: flex;
  flex-grow: 1;
  justify-content: space-between;
}

.metrics {
  display: flex;
  flex-direction: column;
  gap: $kui-space-40;
}

.metric-row {
  align-items: center;
  display: flex;
  gap: $kui-space-30;
}



.metric-separator {
  color: $kui-color-text-neutral;
  margin: 0 $kui-space-40;
}

.service-avatar {
  border-radius: $kui-border-radius-circle;
  height: $kui-space-100;
  width: $kui-space-100;
}

.service-avatars {
  align-self: flex-end;
  display: flex;
  flex-shrink: 0;
  margin-bottom: $kui-space-20;
  margin-left: -$kui-space-100;
}

.service-avatars :deep(.avatar:not(:first-child)){
  margin-left:-$kui-space-50;
  z-index: 1;
}

.not-configured-message {
  align-items: center;
  display: flex;
}

.message-text {
  color: $kui-color-text-neutral;
  font-size: $kui-font-size-30;
  font-weight: $kui-font-weight-regular;
  line-height: $kui-line-height-30;
}

@media (max-width: $kui-breakpoint-phablet) {
  .service-card {
    :deep(.k-card-header),
    :deep(.k-card-body),
    :deep(.k-card-actions) {
      padding: $kui-space-70;
    }

    :deep(.k-card-body) {
      padding-bottom: 0;
    }

    :deep(.k-card-actions) {
      padding-top: $kui-space-60;
    }
  }

  .service-title {
    font-size: $kui-font-size-80;
    line-height: $kui-line-height-80;
  }

  .additional-info {
    align-items: flex-start;
    flex-direction: column;
    gap: $kui-space-70;
  }
}

.metric-dot {
  background-color: $kui-icon-color-success;
  border-radius: $kui-border-radius-circle;
  display: inline-block;
  flex-shrink: 0;
  height: 8px;
  width: 8px;
}
</style>