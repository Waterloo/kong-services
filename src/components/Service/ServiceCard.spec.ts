import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import ServiceCard from './ServiceCard.vue'
import type { Service } from '@/types/models/services'

describe('ServiceCard', () => {
  // Factory function to create test services with sensible defaults
  const createMockService = (overrides: Partial<Service> = {}): Service => ({
    id: 'test-service-1',
    name: 'Test Service',
    description: 'Test Description',
    type: 'REST',
    published: true,
    configured: true,
    versions: [
      {
        id: 'version-1',
        name: '1.0.0',
        description: 'Initial version',
        developer: {
          id: 'dev-1',
          name: 'John Doe',
          email: 'john@example.com',
          avatar: 'https://example.com/avatar1.jpg',
        },
        updated_at: new Date('2024-01-01').toISOString(),
      },
    ],
    metrics: {
      latency: 0.5,
      uptime: 99.5,
      requests: 1000000,
      errors: 0.05,
    },
    ...overrides,
  })

  describe('Snapshot Tests', () => {
    it('matches snapshot for configured published service', () => {
      const service = createMockService({
        published: true,
        configured: true,
      })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches snapshot for unconfigured service', () => {
      const service = createMockService({ configured: false, metrics: undefined })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches snapshot for unpublished service', () => {
      const service = createMockService({ published: false })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      expect(wrapper.html()).toMatchSnapshot()
    })
  })


  describe('Status Configuration', () => {
    it('shows published status for published and configured service', () => {
      const service = createMockService({ published: true, configured: true })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      expect(wrapper.text()).toContain('Published to portal')
    })

    it('shows unpublished status for configured but unpublished service', () => {
      const service = createMockService({
        published: false,
        configured: true,
      })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      expect(wrapper.text()).toContain('Unpublished')
    })

    it('shows not configured status for non-configured service', () => {
      const service = createMockService({ configured: false })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      expect(wrapper.text()).toContain('Not configured with runtime yet')
    })
  })

  describe('Metrics Display', () => {
    it('displays metrics when service is configured', () => {
      const service = createMockService({
        configured: true,
        metrics: {
          latency: 0.45,
          uptime: 99.87,
          requests: 5432100,
          errors: 0.12,
        },
      })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      // Check that metrics are rendered (values will be formatted)
      expect(wrapper.html()).toContain('0.45') // latency
      expect(wrapper.html()).toContain('99.87%') // uptime
      expect(wrapper.html()).toContain('0.12%') // errors
    })

    it('does not display metrics when service is not configured', () => {
      const service = createMockService({
        configured: false,
        metrics: undefined,
      })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      expect(wrapper.find('.metrics').exists()).toBe(false)
      expect(wrapper.find('.not-configured-message').exists()).toBe(true)
    })

    it('formats large request numbers correctly', () => {
      const service = createMockService({
        configured: true,
        metrics: {
          latency: 0.5,
          uptime: 99.5,
          requests: 1234567,
          errors: 0.5,
        },
      })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      // Number should be formatted in compact notation (e.g., "1.2M" or "1M")
      const text = wrapper.text()
      expect(text).toMatch(/1\.2M|1M/)
    })
  })

  describe('Avatar Computation Logic', () => {
    it('shows all avatars when 2 or fewer developers', () => {
      const service = createMockService({
        versions: [
          {
            id: 'v1',
            name: '1.0.0',
            description: 'Version 1',
            developer: {
              id: 'd1',
              name: 'Dev 1',
              email: 'dev1@example.com',
              avatar: 'avatar1.jpg',
            },
            updated_at: new Date().toISOString(),
          },
          {
            id: 'v2',
            name: '2.0.0',
            description: 'Version 2',
            developer: {
              id: 'd2',
              name: 'Dev 2',
              email: 'dev2@example.com',
              avatar: 'avatar2.jpg',
            },
            updated_at: new Date().toISOString(),
          },
        ],
      })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      const avatars = wrapper.findAll('.avatar')
      expect(avatars).toHaveLength(2)
    })

    it('shows +N indicator with 2 avatars when more than 2 developers', () => {
      const service = createMockService({
        versions: [
          {
            id: 'v1',
            name: '1.0.0',
            description: 'V1',
            developer: {
              id: 'd1',
              name: 'Dev 1',
              email: 'd1@example.com',
              avatar: 'avatar1.jpg',
            },
            updated_at: new Date().toISOString(),
          },
          {
            id: 'v2',
            name: '2.0.0',
            description: 'V2',
            developer: {
              id: 'd2',
              name: 'Dev 2',
              email: 'd2@example.com',
              avatar: 'avatar2.jpg',
            },
            updated_at: new Date().toISOString(),
          },
          {
            id: 'v3',
            name: '3.0.0',
            description: 'V3',
            developer: {
              id: 'd3',
              name: 'Dev 3',
              email: 'd3@example.com',
              avatar: 'avatar3.jpg',
            },
            updated_at: new Date().toISOString(),
          },
          {
            id: 'v4',
            name: '4.0.0',
            description: 'V4',
            developer: {
              id: 'd4',
              name: 'Dev 4',
              email: 'd4@example.com',
              avatar: 'avatar4.jpg',
            },
            updated_at: new Date().toISOString(),
          },
        ],
      })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      const avatars = wrapper.findAll('.avatar')
      // Should show 3 avatars: +2 indicator, and first 2 developer avatars
      expect(avatars).toHaveLength(3)

      // First avatar should be the "+2" indicator
      expect(avatars[0].text()).toBe('+2')
    })

    it('deduplicates developer avatars across versions', () => {
      const service = createMockService({
        versions: [
          {
            id: 'v1',
            name: '1.0.0',
            description: 'V1',
            developer: {
              id: 'd1',
              name: 'Same Dev',
              email: 'dev@example.com',
              avatar: 'same-avatar.jpg',
            },
            updated_at: new Date().toISOString(),
          },
          {
            id: 'v2',
            name: '2.0.0',
            description: 'V2',
            developer: {
              id: 'd1',
              name: 'Same Dev',
              email: 'dev@example.com',
              avatar: 'same-avatar.jpg', // Same avatar
            },
            updated_at: new Date().toISOString(),
          },
          {
            id: 'v3',
            name: '3.0.0',
            description: 'V3',
            developer: {
              id: 'd2',
              name: 'Different Dev',
              email: 'other@example.com',
              avatar: 'other-avatar.jpg',
            },
            updated_at: new Date().toISOString(),
          },
        ],
      })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      const avatars = wrapper.findAll('.avatar')
      // Should only show 2 unique avatars, not 3
      expect(avatars).toHaveLength(2)
    })

    it('filters out versions without developer avatars', () => {
      const service = createMockService({
        versions: [
          {
            id: 'v1',
            name: '1.0.0',
            description: 'V1',
            developer: {
              id: 'd1',
              name: 'Dev 1',
              email: 'd1@example.com',
              avatar: 'avatar1.jpg',
            },
            updated_at: new Date().toISOString(),
          },
          {
            id: 'v2',
            name: '2.0.0',
            description: 'V2',
            developer: {
              id: 'd2',
              name: 'Dev 2',
              email: 'd2@example.com',
              avatar: '', // Empty avatar
            },
            updated_at: new Date().toISOString(),
          },
        ],
      })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      const avatars = wrapper.findAll('.avatar')
      // Should only show 1 avatar (the one with valid avatar URL)
      expect(avatars).toHaveLength(1)
    })

    it('correctly calculates +N for many developers', () => {
      // Create 5 developers
      const versions = Array.from({ length: 5 }, (_, i) => ({
        id: `v${i}`,
        name: `${i}.0.0`,
        description: `Version ${i}`,
        developer: {
          id: `d${i}`,
          name: `Dev ${i}`,
          email: `dev${i}@example.com`,
          avatar: `avatar${i}.jpg`,
        },
        updated_at: new Date().toISOString(),
      }))

      const service = createMockService({ versions })
      const wrapper = mount(ServiceCard, {
        props: { service },
      })

      const avatars = wrapper.findAll('.avatar')
      expect(avatars).toHaveLength(3)

      // Should show +3 (5 total - 2 shown = 3 hidden)
      expect(avatars[0].text()).toBe('+3')
    })
  })

})
