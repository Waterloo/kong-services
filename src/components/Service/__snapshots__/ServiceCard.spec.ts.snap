// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServiceCard > Snapshot Tests > matches snapshot for configured published service 1`] = `
"<div data-v-ee47edc8="" data-v-64522201="" class="k-card service-card" aria-label="Open Test Service" role="button" tabindex="0">
  <div data-v-ee47edc8="" class="card-header">
    <div data-v-ee47edc8="" class="card-title">
      <div data-v-64522201="" class="card-header">
        <div data-v-64522201="" class="status-badge"><span data-v-64522201="" class="kui-icon check-small-icon" data-testid="kui-icon-wrapper-check-small-icon" style="box-sizing: border-box; color: rgb(0, 125, 96); display: block; flex-shrink: 0; height: 24px; line-height: 0; width: 24px;"><svg data-testid="kui-icon-svg-check-small-icon" fill="none" height="100%" role="img" viewBox="0 0 24 24" width="100%" xmlns="http://www.w3.org/2000/svg"><path d="M8 14.4L4 10.4L5.4 9L8 11.6L14.6 5L16 6.4L8 14.4Z" fill="currentColor"></path></svg></span><span data-v-64522201="" class="status-text">Published to portal</span></div>
        <div data-v-56e217c6="" data-v-64522201="" class="k-badge info medium">
          <div data-v-56e217c6="" class="badge-content">
            <!---->
            <div data-v-56e217c6="" class="badge-content-wrapper">1 Versions </div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div data-v-ee47edc8="" class="card-content">
    <div data-v-64522201="" class="card-content">
      <h1 data-v-64522201="" class="service-title">Test Service</h1>
      <p data-v-64522201="" class="service-description">Test Description</p>
    </div>
  </div>
  <div data-v-ee47edc8="" class="card-footer">
    <div data-v-64522201="" class="additional-info">
      <div data-v-64522201="" class="metrics">
        <div data-v-64522201="" class="metric-row">
          <div data-v-c143d92f="" data-v-64522201="" class="metric"><span data-v-c143d92f="" class="dot metric-dot"></span> <span data-v-c143d92f="" class="metric-value">0.5</span><span data-v-c143d92f="" class="metric-label">latency</span></div>
        </div>
        <div data-v-64522201="" class="metric-row">
          <div data-v-c143d92f="" data-v-64522201="" class="metric"><span data-v-c143d92f="" class="dot metric-dot"></span> <span data-v-c143d92f="" class="metric-value">99.50%</span><span data-v-c143d92f="" class="metric-label">uptime</span></div>
        </div>
        <div data-v-64522201="" class="metric-row">
          <div data-v-c143d92f="" data-v-64522201="" class="metric"><span data-v-c143d92f="" class="dot metric-dot"></span> <span data-v-c143d92f="" class="metric-value">1M</span><span data-v-c143d92f="" class="metric-label">requests</span></div>
          <div data-v-c143d92f="" data-v-64522201="" class="metric"><span data-v-c143d92f="" class="dot metric-dot"></span> <span data-v-c143d92f="" class="metric-value">0.05%</span><span data-v-c143d92f="" class="metric-label">errors</span></div>
        </div>
      </div>
      <div data-v-64522201="" class="service-avatars">
        <div data-v-dad02320="" data-v-64522201="" class="avatar" style="background-color: rgb(224, 228, 234);"><img data-v-dad02320="" alt="Avatar" src="https://example.com/avatar1.jpg"></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`ServiceCard > Snapshot Tests > matches snapshot for unconfigured service 1`] = `
"<div data-v-ee47edc8="" data-v-64522201="" class="k-card service-card" aria-label="Open Test Service" role="button" tabindex="0">
  <div data-v-ee47edc8="" class="card-header">
    <div data-v-ee47edc8="" class="card-title">
      <div data-v-64522201="" class="card-header">
        <div data-v-64522201="" class="status-badge"><span data-v-64522201="" class="kui-icon progress-icon" data-testid="kui-icon-wrapper-progress-icon" style="box-sizing: border-box; color: rgb(255, 196, 0); display: block; flex-shrink: 0; height: 24px; line-height: 0; width: 24px;"><svg data-testid="kui-icon-svg-progress-icon" fill="none" height="100%" role="img" viewBox="0 0 24 24" width="100%" xmlns="http://www.w3.org/2000/svg"><style>@keyframes kong-icon-spin { from {transform: rotate(0deg);} to {transform: rotate(360deg);} }</style>
<path d="M12 22C10.6333 22 9.34167 21.7375 8.125 21.2125C6.90833 20.6875 5.84583 19.9708 4.9375 19.0625C4.02917 18.1542 3.3125 17.0917 2.7875 15.875C2.2625 14.6583 2 13.3667 2 12C2 10.6167 2.2625 9.32083 2.7875 8.1125C3.3125 6.90417 4.02917 5.84583 4.9375 4.9375C5.84583 4.02917 6.90833 3.3125 8.125 2.7875C9.34167 2.2625 10.6333 2 12 2C12.2833 2 12.5208 2.09583 12.7125 2.2875C12.9042 2.47917 13 2.71667 13 3C13 3.28333 12.9042 3.52083 12.7125 3.7125C12.5208 3.90417 12.2833 4 12 4C9.78333 4 7.89583 4.77917 6.3375 6.3375C4.77917 7.89583 4 9.78333 4 12C4 14.2167 4.77917 16.1042 6.3375 17.6625C7.89583 19.2208 9.78333 20 12 20C14.2167 20 16.1042 19.2208 17.6625 17.6625C19.2208 16.1042 20 14.2167 20 12C20 11.7167 20.0958 11.4792 20.2875 11.2875C20.4792 11.0958 20.7167 11 21 11C21.2833 11 21.5208 11.0958 21.7125 11.2875C21.9042 11.4792 22 11.7167 22 12C22 13.3667 21.7375 14.6583 21.2125 15.875C20.6875 17.0917 19.9708 18.1542 19.0625 19.0625C18.1542 19.9708 17.0958 20.6875 15.8875 21.2125C14.6792 21.7375 13.3833 22 12 22Z" fill="currentColor" style="transform-origin: 50% 50%; animation: kong-icon-spin 1.5s linear infinite;"></path></svg></span><span data-v-64522201="" class="status-text">Not configured with runtime yet</span></div>
        <div data-v-56e217c6="" data-v-64522201="" class="k-badge info medium">
          <div data-v-56e217c6="" class="badge-content">
            <!---->
            <div data-v-56e217c6="" class="badge-content-wrapper">1 Versions </div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div data-v-ee47edc8="" class="card-content">
    <div data-v-64522201="" class="card-content">
      <h1 data-v-64522201="" class="service-title">Test Service</h1>
      <p data-v-64522201="" class="service-description">Test Description</p>
    </div>
  </div>
  <div data-v-ee47edc8="" class="card-footer">
    <div data-v-64522201="" class="not-configured-message"><span data-v-64522201="" class="message-text"><span data-v-64522201="" class="dot"></span> Not configured with runtime yet</span></div>
  </div>
</div>"
`;

exports[`ServiceCard > Snapshot Tests > matches snapshot for unpublished service 1`] = `
"<div data-v-ee47edc8="" data-v-64522201="" class="k-card service-card" aria-label="Open Test Service" role="button" tabindex="0">
  <div data-v-ee47edc8="" class="card-header">
    <div data-v-ee47edc8="" class="card-title">
      <div data-v-64522201="" class="card-header">
        <div data-v-64522201="" class="status-badge"><span data-v-64522201="" class="kui-icon close-icon" data-testid="kui-icon-wrapper-close-icon" style="box-sizing: border-box; color: rgb(108, 116, 137); display: block; flex-shrink: 0; height: 24px; line-height: 0; width: 24px;"><svg data-testid="kui-icon-svg-close-icon" fill="none" height="100%" role="img" viewBox="0 0 24 24" width="100%" xmlns="http://www.w3.org/2000/svg"><path d="M6.4 19L5 17.6L10.6 12L5 6.4L6.4 5L12 10.6L17.6 5L19 6.4L13.4 12L19 17.6L17.6 19L12 13.4L6.4 19Z" fill="currentColor"></path></svg></span><span data-v-64522201="" class="status-text">Unpublished</span></div>
        <div data-v-56e217c6="" data-v-64522201="" class="k-badge info medium">
          <div data-v-56e217c6="" class="badge-content">
            <!---->
            <div data-v-56e217c6="" class="badge-content-wrapper">1 Versions </div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div data-v-ee47edc8="" class="card-content">
    <div data-v-64522201="" class="card-content">
      <h1 data-v-64522201="" class="service-title">Test Service</h1>
      <p data-v-64522201="" class="service-description">Test Description</p>
    </div>
  </div>
  <div data-v-ee47edc8="" class="card-footer">
    <div data-v-64522201="" class="additional-info">
      <div data-v-64522201="" class="metrics">
        <div data-v-64522201="" class="metric-row">
          <div data-v-c143d92f="" data-v-64522201="" class="metric"><span data-v-c143d92f="" class="dot metric-dot"></span> <span data-v-c143d92f="" class="metric-value">0.5</span><span data-v-c143d92f="" class="metric-label">latency</span></div>
        </div>
        <div data-v-64522201="" class="metric-row">
          <div data-v-c143d92f="" data-v-64522201="" class="metric"><span data-v-c143d92f="" class="dot metric-dot"></span> <span data-v-c143d92f="" class="metric-value">99.50%</span><span data-v-c143d92f="" class="metric-label">uptime</span></div>
        </div>
        <div data-v-64522201="" class="metric-row">
          <div data-v-c143d92f="" data-v-64522201="" class="metric"><span data-v-c143d92f="" class="dot metric-dot"></span> <span data-v-c143d92f="" class="metric-value">1M</span><span data-v-c143d92f="" class="metric-label">requests</span></div>
          <div data-v-c143d92f="" data-v-64522201="" class="metric"><span data-v-c143d92f="" class="dot metric-dot"></span> <span data-v-c143d92f="" class="metric-value">0.05%</span><span data-v-c143d92f="" class="metric-label">errors</span></div>
        </div>
      </div>
      <div data-v-64522201="" class="service-avatars">
        <div data-v-dad02320="" data-v-64522201="" class="avatar" style="background-color: rgb(224, 228, 234);"><img data-v-dad02320="" alt="Avatar" src="https://example.com/avatar1.jpg"></div>
      </div>
    </div>
  </div>
</div>"
`;
